src/library_html5.js

```diff
@@ -935,17 +935,41 @@ var LibraryHTML5 = {
 
   $fillOrientationChangeEventData__deps: ['$screenOrientation'],
   $fillOrientationChangeEventData: (eventStruct) => {
-    var orientations  = ["portrait-primary", "portrait-secondary", "landscape-primary", "landscape-secondary"];
-    var orientations2 = ["portrait",         "portrait",           "landscape",         "landscape"];
-
-    var orientationString = screenOrientation();
-    var orientation = orientations.indexOf(orientationString);
-    if (orientation == -1) {
-      orientation = orientations2.indexOf(orientationString);
+    // OrientationType enum
+    var orientationsType1 = ['portrait-primary', 'portrait-secondary', 'landscape-primary', 'landscape-secondary'];
+    // alternative selection from OrientationLockType enum
+    var orientationsType2 = ['portrait',         'portrait',           'landscape',         'landscape'];
+
+    var orientationIndex = -1;
+    var orientationAngle =  0;
+    var screenOrientObj  = screenOrientation();
+    if (screenOrientObj) {
+      var orientationType;
+      if (typeof screenOrientObj === 'object') {
+        orientationType  = screenOrientObj.type;
+        orientationAngle = screenOrientObj.angle;
+      } else {
+        // Edge only supports the older string type until 2020 (pre-Chrome)
+        orientationType  = String(screenOrientObj);
+      }
+      orientationIndex = orientationsType1.indexOf(orientationType);
+      if (orientationIndex == -1) {
+        orientationIndex = orientationsType2.indexOf(orientationType);
+      }
+      if (orientationIndex != -1) {
+        orientationIndex = 1 << orientationIndex;
+      }
+    } else {
+#if MIN_SAFARI_VERSION < 0x100400
+      // fallback for Safari earlier than 16.4 (March 2023)
+      if (window && (window['orientation'] !== undefined)) {
```

Comment:
`if (window)` will not work, but will throw an exception if `window` is not defined. I think this `else block could just be

```js
} else {
  orientationAngle = window.orientation; // Safari earlier than 16.4 (March 2023)
}
```
as `makeSetValue()` will coerce any undefineds into writing a zero.